src/main/java/com/mraof/minestuck/block/redstone/AreaEffectBlock.java

```diff
@@ -79,27 +79,26 @@ public <T extends BlockEntity> BlockEntityTicker<T> getTicker(Level level, Block
 	@Override
 	public InteractionResult use(BlockState state, Level level, BlockPos pos, Player player, InteractionHand hand, BlockHitResult hit)
 	{
-		if(player.isCreative() && !CreativeShockEffect.doesCreativeShockLimit(player, CreativeShockEffect.LIMIT_MACHINE_INTERACTIONS))
+		if(!canInteract(player) || !(level.getBlockEntity(pos) instanceof AreaEffectBlockEntity be))
+			return InteractionResult.PASS;
+		
+		ItemStack heldItemStack = player.getItemInHand(hand);
+		
+		if(heldItemStack.getItem() instanceof PotionItem)
+		{
+			clickWithPotion(level, pos, player, be, heldItemStack);
+		} else
 		{
-			if(level.getBlockEntity(pos) instanceof AreaEffectBlockEntity be)
-			{
-				
-				ItemStack heldItemStack = player.getItemInHand(hand);
-				
-				if(heldItemStack.getItem() instanceof PotionItem)
-				{
-					clickWithPotion(level, pos, player, be, heldItemStack);
-				} else
-				{
-					if(level.isClientSide)
-						MSScreenFactories.displayAreaEffectScreen(be);
-				}
-				
-				return InteractionResult.sidedSuccess(level.isClientSide);
-			}
+			if(level.isClientSide)
+				MSScreenFactories.displayAreaEffectScreen(be);
 		}
 		
-		return InteractionResult.PASS;
+		return InteractionResult.sidedSuccess(level.isClientSide);
+	}
+	
+	public static boolean canInteract(Player player)
+	{
+		return player.isCreative() && !CreativeShockEffect.doesCreativeShockLimit(player, CreativeShockEffect.LIMIT_MACHINE_INTERACTIONS);
```

Comment:
Seems to make sense to have canInteract() be in CreativeShockEffect, generalized so it can be used by all the other variants of canInteract() dispersed throughout the project

Reply:
I'm not so sure about that myself. I can't think of a good reason to do it in this circumstance.
### Moving/combining the functions
There is a slight difference between the versions of `canInteract()` already (some require creative, others don't). Even if some are the same, I would want to treat them as if they only happen to be the same, rather than that they must be the same.
### Extracting part of the functions
And since they're so small already, I don't think there's much room to extract common parts to create simpler code. Depending on what exactly is being extracted, it may also obscure what the function actually checks.

Does that cover what you were considering, or do you have a different angle?